mernarezk/assignment4/CountIslandsTest.java

```diff
@@ -0,0 +1,23 @@
+package Assignment4;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+class CountIslandsTest {
+
+	@Test
+	void test() {
+		CountIslands tester = new CountIslands();
+		boolean grid[][] = new boolean[][] { { false, true, false, true }, { true, true, false, false },
+				{ false, false, true, false }, { false, false, true, false } };
+		assertEquals(tester.countIslands(grid, 4, 4), 3);
+
+		boolean allWater[][] = new boolean[][] { { false, false }, { false, false } };
+		assertEquals(tester.countIslands(allWater, 2, 2), 0);
+
+		boolean oneLand[][] = new boolean[][] { { true, true }, { true, true } };

```

Comment:
Please add more test cases, like: a few islands of the diagonal (check that diagonal connections count as separate islands), chequered grid, islands only in corners, a big island with a lake in the middle, L shaped islands, U shaped islands, Z shape (2 islands on top and bottom plus a few islands from the diagonal) etc.

Reply:
done! :)